certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/Main.kt

```diff
@@ -0,0 +1,168 @@
+  var output: File? = null
+
+  @Parameters(paramLabel = "file", description = ["Input File"], arity = "0..1")
+  var file: String? = null
+
+  override fun call(): Int = try {
+    if (host != null) {
+      queryHost()
+    } else if (file != null) {
+      showPemFile()
+    }
+    0
+  } catch (ce: CertificationException) {
+    System.err.println("Error: ${Ansi.AUTO.string(" @|yellow ${ce.message}|@")}")
+    if (verbose) {
+      ce.cause?.printStackTrace()
+    }
+    -2
+  } catch (ue: UsageException) {
+    System.err.println("Error: ${Ansi.AUTO.string(" @|red ${ue.message}|@")}")
+    if (verbose) {
+      ue.cause?.printStackTrace()
+    }
+    -1
+  }
+
+  private fun showPemFile() {
+    val certificate = parsePemCertificate(File(file!!))
+    println(certificate.prettyPrintCertificate())
+  }
+
+  private fun queryHost() {
+    val x509certificates = fromHttps(host!!)
+    prettyPrintChain(x509certificates)
+
+    if (output != null) {
+      try {
+        outputCertificates(output!!, x509certificates)
+      } catch (ioe: IOException) {
+        throw UsageException("Unable to write to $output", ioe)
+      }
+    }
+  }
+
+  private fun outputCertificates(
+    output: File,
+    certificates: List<X509Certificate>
+  ) {
+    when {
+      output.isDirectory -> certificates.forEach {
+        outputCertificate(File(output, "${it.sha256Hash().hex()}.pem"), it)
+      }
+      else -> outputCertificate(output, certificates.first())
+    }
+  }
+
+  private fun outputCertificate(
+    output: File,
+    certificate: X509Certificate
+  ) {
+    output.writeText(certificate.certificatePem())
+  }
+
+  private fun prettyPrintChain(certificates: List<X509Certificate>) {
+    certificates.forEachIndexed { i, certificate ->
+      if (i > 0) {
+        println()
+      }
+
+      val certifikit = CertificateAdapters.certificate.fromDer(certificate.encoded.toByteString())
+      println(certifikit.prettyPrintCertificate())
+    }
+  }
+
+  private fun parsePemCertificate(file: File): Certificate {
+    try {
+      val data = file.readText()
+          .replace("-----BEGIN CERTIFICATE-----\n", "")
+          .replace("-----END CERTIFICATE-----\n", "")
+          .decodeBase64()!!
+      val certificate = CertificateAdapters.certificate.fromDer(data)
+      return certificate
+    } catch (fnfe: FileNotFoundException) {
+      throw UsageException("No such file: $file", fnfe)
+    }
+  }
+
+  class VersionProvider : IVersionProvider {
+    override fun getVersion(): Array<String> {
+      return arrayOf("$NAME ${versionString()}")
+    }
+  }
+
+  companion object {
+    internal const val NAME = "cft"
+
+    @JvmStatic
+    fun main(args: Array<String>) {
+      exitProcess(CommandLine(Main()).execute(*args))
+    }
+
+    private fun versionString(): String? {
+      val prop = Properties()
+      Main::class.java.getResourceAsStream("/certifikit-version.properties")
```

Comment:
Wanna do like we do in OkHttp, and make a constant, `Certifikit.VERSION` ?

Reply:
Will follow up